SO-3380: Export script

zoliMolnar opened the pull request:
This pull request adds a script which can initiate an export from a server with an installed Snow Owl instance on it.

Commit:
SO-3380: Initial version of the snowowl_export.sh script

cmark commented on releng/com.b2international.snowowl.server.update/assembly/common/scripts/snowowl_export.sh:
```diff
@@ -0,0 +1,176 @@
+#!/usr/bin/env bash
+#
+# Copyright 2018 B2i Healthcare Pte Ltd, http://b2i.sg
```
Fix copyright year!

cmark commented on releng/com.b2international.snowowl.server.update/assembly/common/scripts/snowowl_export.sh:
```diff
@@ -0,0 +1,176 @@
+#
+
+#
+# Snow Owl terminology server deploy script
```
⚠️ Copy-paste!

cmark commented on releng/com.b2international.snowowl.server.update/assembly/common/scripts/snowowl_export.sh:
```diff
@@ -0,0 +1,176 @@
+EXPORT_UUID=""
+
+# Media type to use for the export configuration.
+MEDIA_TYPE="application/vnd.com.b2international.snowowl+json"
```
Split variables into two named sections, one is for actual configuration parameters with either default or no value and the other for global variables/constants.

cmark commented on releng/com.b2international.snowowl.server.update/assembly/common/scripts/snowowl_export.sh:
```diff
@@ -0,0 +1,176 @@
+ACCEPT_HEADER="application/octet-stream"
+
+# The address where the export config endpoint can be found.
+EXPORT_CONFIG_POST_ENDPOINT="snowowl/snomed-ct/v3/exports"
```
Snow Owl actual API endpoint should be configurable via a `SNOWOWL_BASE_URL` variable (to support earlier versions of the export API).

cmark commented on releng/com.b2international.snowowl.server.update/assembly/common/scripts/snowowl_export.sh:
```diff
@@ -0,0 +1,176 @@
+TARGET_FOLDER=""
+
+# The server where the export should be initiated on
+TARGET_ENVIRONMENT=""
```
Is this the base URL variable? If so, please rename it. See another comment below.

cmark commented on releng/com.b2international.snowowl.server.update/assembly/common/scripts/snowowl_export.sh:
```diff
@@ -0,0 +1,176 @@
+RENAMED_EXPORT_FILE="snow_owl_delta_export"
+
+# Calculates the current location of this specific bash script (works until the path is not a symlink)
+SCRIPT_LOCATION="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
```
This will always override the user-specified `TARGET_FOLDER` value, which can result in unwanted download locations. Allow users to specify the output directory and remove this variable.

cmark commented on releng/com.b2international.snowowl.server.update/assembly/common/scripts/snowowl_export.sh:
```diff
@@ -0,0 +1,176 @@
+EXPORT_FILE_NAME="snow_owl_delta_export"
+
+# The renamed version of the export file.
+RENAMED_EXPORT_FILE="snow_owl_delta_export"
```
Looks like this variable is unnecessary, please remove it.

cmark commented on releng/com.b2international.snowowl.server.update/assembly/common/scripts/snowowl_export.sh:
```diff
@@ -0,0 +1,176 @@
+
+    echo "Downloading snapshot export with UUID: ${EXPORT_DOWNLOAD_GET_ENDPOINT}"
+
+    curl -u "${SNOW_OWL_USER}:${SNOW_OWL_PASSWORD}" -X GET -H "Accept: ${ACCEPT_HEADER}" -ko "${EXPORT_FILE_NAME}" "${EXPORT_DOWNLOAD_GET_ENDPOINT}" 
```
You can immediately download the file with a timestamped name with `"${EXPORT_FILE_NAME}_${DATE}"`

cmark commented on releng/com.b2international.snowowl.server.update/assembly/common/scripts/snowowl_export.sh:
```diff
@@ -0,0 +1,176 @@
+    echo "EXPORT_UUID: ${EXPORT_UUID}"
+    EXPORT_DOWNLOAD_GET_ENDPOINT="${EXPORT_UUID}/archive"
+
+    echo "Downloading snapshot export with UUID: ${EXPORT_DOWNLOAD_GET_ENDPOINT}"
```
`Snapshot` is used in the log output, I guess it should print the currently configured RF2 export type.

Commit:
SO-3380: Make export type configurable, add base URL configurable var...

... add support for saving to predetermined folder, fix comments,
separate variables

Commit:
SO-3380: Remove target environment variable add base_url and api_url

cmark commented on releng/com.b2international.snowowl.server.update/assembly/common/scripts/snowowl_export.sh:
```diff
@@ -0,0 +1,200 @@
+SNOW_OWL_BASE_URL="http://localhost:8080"
+
+# Api URL of the running Snow Owl server.
+SNOW_OWL_API_URL="snowowl/snomed-ct/v3"
```
⚠️ Missing slash from the beginning of the default value!

Commit:
SO-3380: Simplify saving of result export

Commit:
SO-3380: Add leading slash to default value of SNOW_OWL_API_URL var

Commit:
SO-3380: Fix echo call, remove unused variable, rename methods

cmark reviewed (approved):
LGTM!

zoliMolnar commented:
Thank you!
